ramblings/holey_domains.jl

```diff
@@ -0,0 +1,111 @@
+# Generating dummy column names & data
+
+inputs = Dict(
+    "scenario" => ["1-squirrels-fight-back", "2-the-squirrels-dont-like-peanut-butter", "3-chaos-with-ed-miliband"],
+
+    "year" => [2020, 1997, 2021, 2043, 2099],
+
+    "mode" => ["Hovercraft", "SUV", "Tricycle"],
+)
+
+variables = ["kgco2perkm", "qualys-change-per-km"]
+
+
+possibilities = []
+for (k,array) in inputs
+    push!(possibilities, ["$k=$v" for v in array])
+end
+
+column_names = []
+combinations = Iterators.product(possibilities...) |> collect
+for var in variables
+    # +5 magic number makes it more likely to use all inputs
+    for i in 1:sqrt(length(combinations))
+        push!(column_names, "$(var)_$(join(rand(rand(combinations),rand(1:(length(inputs)+5))) |> unique, "_"))")
+    end
+end
+
+using DataFrames
+df = DataFrame(hcat(rand(1:100,100,2), rand(100,length(column_names))), ["Origin", "Destination", column_names...])
+
+# There is some ambiguity here
+# if we have e.g. year=2099_mode=SUV and mode=SUV_year=2099 there's clearly a problem
+# ditto for if we have a column for year=2099 and then also a column for year=2099_mode=suv
+#
+# Also ditto for ambiguous origin->dest entries
+# 
+# I think it's probably OK to assume that the real data is cleaner than that, though, and just ignore it?
+
+
+# "Real" code hereon - this stuff could be useful for real, doesn't just generate dummy data
```

Comment:
NB: if we provide all the raw column names to JS (which should always be pretty small), there's no reason why this code can't be in JavaScript in the frontend which could help responsiveness